rust/libs/connlib/phoenix-channel/src/lib.rs

```diff
@@ -228,6 +228,40 @@ impl std::error::Error for InternalError {
     }
 }
 
+/// Parses a Retry-After header value into a Duration.
+///
+/// The header can be either:
+/// - A number of seconds (e.g., "120")
+/// - An HTTP date in IMF-fixdate format (e.g., "Wed, 21 Oct 2015 07:28:00 GMT")
+fn parse_retry_after(response: &tungstenite::http::Response<Option<Vec<u8>>>) -> Option<Duration> {
+    let header = response.headers().get("retry-after")?;
+    let value = header.to_str().ok()?;
+    parse_retry_after_value(value)
```

Comment:
Nit: I think this is cleaner to write as:


```suggestion
    let duration= parse_retry_after_value(value)?;
    
    Some(duration)
```